docs/developers/tutorials/dex-walkthrough.md

```diff
@@ -0,0 +1,920 @@
+---
+id: dex-walkthrough
+title: DEX Walkthrough
+---
+
+[comment]: # (mx-abstract)
+
+## Introduction
+
+If you are building a project that involves decentralized exchange functionality, integrating DEX contracts can be a crucial step in achieving your goals. These contracts provide the underlying infrastructure necessary to facilitate the exchange of assets on a decentralized platform. However, implementing these contracts can be a complex process, and understanding the various public endpoints and functions can be challenging. In this in-depth walkthrough, we will guide you through the process of integrating DEX contracts into your __MultiversX__ project. We will cover all of the main contracts involved in a typical DEX implementation, and provide detailed explanations of the most commonly used public endpoints. By the end of this tutorial, you should have a solid understanding of how to implement DEX functionality in your own project, and be able to make informed decisions about how to customize and extend the functionality to meet your specific needs.
+
+[comment]: # (mx-context-auto)
+
+## Prerequisites
+
+The DEX contracts are a bit more advance than you standard SCs, so basic knowledge about Rust SC development is required. If you are a beginner, an easier starting point (like the Crowdfunding SC or the Staking SC tutorials) is strongly advised. Also, to better grasp the DEX contracts implementation, it is important that you first understand the xExchange economic model.
+
+:::important
+You can find the xExchange whitepaper here:
+https://xexchange.com/x-exchange-economics.pdf
+:::
+
+[comment]: # (mx-context-auto)
+
+## **DEX repo structure** & recommendations
+
+The main DEX contracts are as follow:
+
```

Comment:
missing router contract

Reply:
This documentation was written with a developer go to summary approach in mind, so I did not cover all the contracts, but only those that are usually used by external developers that are trying to build on top of the DEX. For that reason, I skipped the router contract, as that one is more of a internal utility contract. That being said, I've just realised that the Router contract also has the `multiPairSwap` endpoint which can indeed be used by other developers. I'll come back with a walkthrough for that endpoint as well.

Reply:
Updated.